src/bin/pg_autoctl/pghba.c

```diff
@@ -172,6 +172,59 @@ pghba_ensure_host_rule_exists(const char *hbaFilePath,
 }
 
 
+/*
+ * write_trust_local_hba_rule gets the hbaFilePath and overrides it with
+ * 		"local all all  trust"
+ * 	which means allow all connections via unix domain sockets.
+ */
+bool
+write_trust_local_hba_rule(const char *hbaFilePath)
+{
+	const char *userName = "all";
+	const char *authenticationScheme = "trust";
+
+	PQExpBuffer hbaLineBuffer = createPQExpBuffer();
+	if (hbaLineBuffer == NULL)
+	{
+		log_error("Failed to allocate memory");
+		return false;
+	}
+
+	appendPQExpBufferStr(hbaLineBuffer, "local ");
+
+	append_database_field(hbaLineBuffer, HBA_DATABASE_ALL, NULL);
+	appendPQExpBufferStr(hbaLineBuffer, " ");
+
+	appendPQExpBufferStr(hbaLineBuffer, userName);
+	appendPQExpBufferStr(hbaLineBuffer, " ");
+
+	appendPQExpBuffer(hbaLineBuffer, " %s", authenticationScheme);
+
+	appendPQExpBufferStr(hbaLineBuffer, HBA_LINE_COMMENT "\n");
```

Comment:
Please re-use `pghba_ensure_host_rule_exists` and extend it with a new parameter `HBAConnectionType` (enum with 6 entries for `local`, `host`, `hostssl`, `hostnossl`, `hostgssenc`, `hostnogssenc` per https://www.postgresql.org/docs/current/auth-pg-hba-conf.html) much like `HBADatabaseType`. This parameter would replace `bool ssl` obviously.